### DISCUSSION TITLE
Welcome to CIPster Discussions!

### DISCUSSION THREAD

**Discussion by liftoff-sr** [liftoff-sr/CIPster#18] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Use Discussions to ask questions, and use Issues to indicate bugs.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by langmuir78** [comment]
There is any plan to add support for the QoS object in the library? 

I imagine that the concrete implementation would strongly depend on the operating system, but what effort would be needed to add the relevant CIP class to your code?

**Reply by liftoff-sr** [reply]
No such plans at this time.

**Comment by liftoff-sr** [comment]
> There is any plan to add support for the QoS object in the library?

I do not have such a plan.


> I imagine that the concrete implementation would strongly depend on the operating system, but what effort would be
> needed to add the relevant CIP class to your code?

Adding a new class is relatively easy, you have to register the class after it is instantiated.  Search the source code
for the register operation, and you may find it being done 8 times or so.  Each of these 8 then becomes and example that
you can study and use as a guide.

**Reply by langmuir78** [reply]
Thank you for the prompt reply.

I'll study check you code and see if this upgrade is within my reach, i'm not at all an expert of the CIP/Ethernet/IP protocol.
